### PR TITLE
Continuous profiling

### PR DESCRIPTION
**Pull request by baurine**
What did:
1. In the FE side, implement the conprof web pages, including:
    - profiles list page
    - profile detail page
    - download all profiles of a certain timestamp
    - directly view a single profile of an instance in the browser
2. In the backend side, add auth check, reverse the APIs to the ng monitoring

Screenshots:
- Side menu
  image

- Changes for manually profiling list page  
  image

- Changes for manually profiling detail page
  image

- Continuous Profling list page
  image

- Continuous Profiling setting page
  image

- Continuous Profilng detail page
  image

- Not deploy ng monitoring
  image

- Not start ng monitoring
  image

Demo video:

https://user-images.githubusercontent.com/1284531/138676016-6c758c5f-9b66-42a8-8e39-2b8518fef46c.mp4

### CODE REVIEW

**Comment by breeswish** [comment]
The code freeze is approaching. What's the status of this PR? Is it working in progress or ready to review?

**Comment by baurine** [comment]
> The code freeze is approaching. What's the status of this PR? Is it working in progress or ready to review?

The remain refinements will commit today and it will become review ready.

File: ui/lib/apps/InstanceProfiling/pages/List.tsx [hunk 1]
```diff
@@ -94,6 +101,15 @@ export default function Page() {
...
     () => [
+      {
+        name: t('instance_profiling.list.table.columns.order'),
+        key: 'order',
+        minWidth: 100,
+        maxWidth: 150,
+        onRender: (_rec, idx) => {
+          return <span>{historyLen - idx}</span>
+        },
+      },
```
[the remark this answers is not in the document]
Comment (baurine) [reply ui/lib/apps/InstanceProfiling/pages/List.tsx]: done

File: pkg/apiserver/profiling/conprof_router.go [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+}
+
+func (s *Service) reverseProxy(targetPath string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		ngMonitoringAddr, err := s.getNgMonitoringAddrFromCache()
+		if err != nil {
+			_ = c.Error(ErrLoadNgMonitoringAddr.Wrap(err, "Load NgMonitoring address failed"))
+			return
+		}
+		if ngMonitoringAddr == NgMonitoringNotDeploy {
```
[the remark this answers is not in the document]
Comment (baurine) [reply pkg/apiserver/profiling/conprof_router.go]: done

File: pkg/apiserver/profiling/conprof_router.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+
+func (s *Service) resolveNgMonitoringAddress() string {
+	pi, err := topology.FetchPrometheusTopology(s.lifecycleCtx, s.params.EtcdClient)
+	if pi == nil || err != nil {
+		return NgMonitoringNotDeploy
+	}
+
+	addr, err := topology.FetchNgMonitoringTopology(s.lifecycleCtx, s.params.EtcdClient)
+	if err == nil && addr != "" {
+		return fmt.Sprintf("http://%s", addr)
```
[the remark this answers is not in the document]
Comment (baurine) [reply pkg/apiserver/profiling/conprof_router.go]: as @crazycs520 said, currently we only support http.

File: ui/lib/apps/InstanceProfiling/pages/List.tsx [hunk 4]
```diff
@@ -183,9 +219,26 @@ export default function Page() {
...
           <Form.Item>
-            <Button type="primary" htmlType="submit" loading={submitting}>
-              {t('instance_profiling.list.control_form.submit')}
-            </Button>
+            {conprofEnable ? (
+              <Tooltip
+                title={t(
+                  'instance_profiling.list.control_form.submit_disable_tooltip'
+                )}
+              >
```
[the remark this answers is not in the document]
Comment (baurine) [reply ui/lib/apps/InstanceProfiling/pages/List.tsx]: done
WX20211031-174448@2x

File: ui/lib/apps/InstanceProfiling/pages/Detail.tsx [hunk 5]
```diff
@@ -142,7 +161,13 @@ export default function Page() {
             {t('instance_profiling.detail.download')}
           </Button>
         }
-      />
+      >
+        {respData && (
+          <DateTime.Calendar
+            unixTimestampMs={respData.task_group_status!.started_at! * 1000}
+          />
+        )}
```
[the remark this answers is not in the document]
Comment (baurine) [reply ui/lib/apps/InstanceProfiling/pages/Detail.tsx]: image

File: ui/lib/apps/ContinuousProfiling/pages/List.tsx [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    () => ngMonitoringConfig?.continuous_profiling?.enable ?? false,
+    [ngMonitoringConfig]
+  )
+  const conProfStatusTooltip = useMemo(() => {
+    if (conprofEnable) {
+      return t('continuous_profiling.list.control_form.enable_tooltip')
+    } else {
+      return t('continuous_profiling.list.control_form.disable_tooltip')
+    }
+  }, [conprofEnable, t])
```
[the remark this answers is not in the document]
Comment (baurine) [reply ui/lib/apps/ContinuousProfiling/pages/List.tsx]: done

WX20211031-170229@2x
WX20211031-170212@2x

**Review by breeswish — APPROVED** [review]
I have not carefully reviewed. Generally looks good.

**Comment by baurine** [comment]
Merge it first, the remaining refinements will be committed as a new PR.

File: ui/lib/apps/InstanceProfiling/pages/List.tsx [hunk 7]
```diff
@@ -110,20 +126,40 @@ export default function Page() {
...
+            )
+          } else if (rec.state === 1) {
+            // running
             return (
               <Badge
                 status="processing"
                 text={t('instance_profiling.list.table.status.running')}
               />
             )
           } else if (rec.state === 2) {
```
[the remark this answers is not in the document]
Comment (baurine) [reply ui/lib/apps/InstanceProfiling/pages/List.tsx:138]: Fix it in PR #1031

**Pull request merged**